packages/anoncreds/src/updates/0.3.1-0.4/credentialExchangeRecord.ts

```diff
@@ -0,0 +1,148 @@
+ *
+ * ```json
+ * {
+ *   "credentials": [
+ *     {
+ *       "credentialRecordId": "credential-id",
+ *       "credentialRecordType": "indy"
+ *     },
+ *     {
+ *       "credentialRecordId": "credential-id2",
+ *       "credentialRecordType": "jsonld"
+ *     }
+ *   ]
+ * }
+ * ```
+ *
+ * Wil be tranformed into the following 0.4 credential record structure (unrelated keys omitted):
+ * ```json
+ * {
+ *   "credentials": [
+ *     {
+ *       "credentialRecordId": "credential-id",
+ *       "credentialRecordType": "anoncreds"
+ *     },
+ *     {
+ *       "credentialRecordId": "credential-id2",
+ *       "credentialRecordType": "jsonld"
+ *     }
+ *   ]
+ * }
+ * ```
+ */
+export function migrateIndyCredentialTypeToAnonCredsCredential<Agent extends BaseAgent>(
+  agent: Agent,
+  credentialRecord: CredentialExchangeRecord
+) {
+  agent.config.logger.debug(
+    `Migrating credential record with id ${credentialRecord.id} to anoncreds credential binding for version 0.4`
+  )
+
+  const INDY_CREDENTIAL_RECORD_TYPE = 'indy'
+  const ANONCREDS_CREDENTIAL_RECORD_TYPE = 'anoncreds'
+
+  for (const credential of credentialRecord.credentials) {
+    if (credential.credentialRecordType === INDY_CREDENTIAL_RECORD_TYPE) {
+      agent.config.logger.debug(`Updating credential binding ${credential.credentialRecordId} to anoncreds type`)
+      credential.credentialRecordType = ANONCREDS_CREDENTIAL_RECORD_TYPE
+    }
+  }
+
+  agent.config.logger.debug(
+    `Successfully migrated credential record with id ${credentialRecord.id} to anoncreds credential binding for version 0.4`
+  )
+}
+
+/**
+ * Migrates the indy credential metadata type to anoncreds credential metadata type.
+ *
+ * The following 0.3.1 creedential metadata structure (unrelated keys omitted):
+ *
+ * ```json
+ * {
+ *   "_internal/indyRequest": {}
+ *   "_internal/indyCredential": {}
+ * }
+ * ```
+ *
+ * Wil be tranformed into the following 0.4 credential metadata structure (unrelated keys omitted):
+ * ```json
+ * {
+ *   "_anoncreds/credentialRequest": {}
+ *   "_anoncreds/credential": {}
+ * }
+ * ```
+ */
+export function migrateIndyCredentialMetadataToAnonCredsMetadata<Agent extends BaseAgent>(
+  agent: Agent,
+  credentialRecord: CredentialExchangeRecord
+) {
+  agent.config.logger.debug(
+    `Migrating credential record with id ${credentialRecord.id} to anoncreds metadata for version 0.4`
+  )
+
+  const INDY_CREDENTIAL_REQUEST_METADATA = '_internal/indyRequest'
+  const INDY_CREDENTIAL_METADATA = '_internal/indyCredential'
+
+  const ANONCREDS_CREDENTIAL_REQUEST_METADATA = '_anoncreds/credentialRequest'
+  const ANONCREDS_CREDENTIAL_METADATA = '_anoncreds/credential'
+
+  const indyCredentialRequestMetadata = credentialRecord.metadata.get(INDY_CREDENTIAL_REQUEST_METADATA)
+  if (indyCredentialRequestMetadata) {
+    // TODO: we if we choose to rename master secret to link secret in anoncreds-rs we should also rename it in the request
```

Comment:
When you call create credential request in indy-sdk / anoncreds rs) it will return the following metadata:

```
{
          "master_secret_blinding_data": {
            "v_prime": "6088566065720309491695644944398283228337587174153857313170975821102428665682789111613194763354086540665993822078019981371868225077833338619179176775427438467982451441607103798898879602785159234518625137830139620180247716943526165654371269235270542103763086097868993123576876140373079243750364373248313759006451117374448224809216784667062369066076812328680472952148248732117690061334364498707450807760707599232005951883007442927332478453073050250159545354197772368724822531644722135760544102661829321297308144745035201971564171469931191452967102169235498946760810509797149446495254099095221645804379785022515460071863075055785600423275733199",
            "vr_prime": null
          },
          "nonce": "131502096406868204437821",
          "master_secret_name": "walletId28c602347-3f6e-429f-93cd-d5aa7856ef3f"
        }
```

If we move to using link secret in AnonCreds RS, we should update all metadata and transform the master_secret keys to link_secret keys in the Indy SDK holder service in AFJ.

Did you also update these fields in your link secret pr in anoncreds rs?